### PR TITLE
[SPARK-43837][SQL] Assign a name to the error class _LEGACY_ERROR_TEMP_103[1-2]

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to assign a name to the error class _LEGACY_ERROR_TEMP_103[1-2].

### Why are the changes needed?
The changes improve the error framework.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Update existed UT.
- Pass GA.

### CODE REVIEW

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -904,6 +904,11 @@
     ],
     "sqlState" : "22003"
   },
+  "INVALID_BOUNDARY" : {
+    "message" : [
+      "Boundary <boundary> is not a valid integer: <value>."
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:909]: Precisely, speaking:
- `start` can be 0, Long.MinValue, or [Int.MinValue, Int.MaxValue]
- `stop` - 0, Long.MaxValue, or [Int.MinValue, Int.MaxValue]

Could you, please, add sub-classes like:
```
  "INVALID_BOUNDARY" : {
    "message" : [
      "The boundary <boundary> is invalid: <invalidValue>."
...
    "subClass" : {
      "START" : {
        "message" : [
          "Expected ... ."
        ]
      },
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala [hunk 2]
```diff
@@ -2877,14 +2877,18 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
 
   def invalidBoundaryStartError(start: Long): Throwable = {
     new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1301",
-      messageParameters = Map("start" -> start.toString))
+      errorClass = "INVALID_BOUNDARY",
+      messageParameters = Map(
+        "boundary" -> "start",
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:2882]: Could you quote it as an id using `toSQLId`

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala [hunk 3]
```diff
@@ -2877,14 +2877,18 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
...
+        "value" -> toSQLValue(start, LongType)))
   }
 
   def invalidBoundaryEndError(end: Long): Throwable = {
     new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1302",
-      messageParameters = Map("end" -> end.toString))
+      errorClass = "INVALID_BOUNDARY",
+      messageParameters = Map(
+        "boundary" -> "end",
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:2890]: Wrap it by `toSQLId()`, please.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala (hunk 2 above)
Comment (panbingkun) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala]: This is done.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala (hunk 3 above)
Comment (panbingkun) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala]: ditto

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Pull request closed without merging**